tuf/client/updater.py

```diff
@@ -1130,9 +1130,11 @@ def _update_root_metadata(self, current_root_metadata):
       None.
     """
 
-    # Retrieve the latest, remote root.json.
-    latest_root_metadata_file = self._get_metadata_file(
-        'root', 'root.json', DEFAULT_ROOT_UPPERLENGTH, None)
+    # Retrieve the latest, remote root.json *WITHOUT* verifying it just yet.
+    # We will verify it soon if all goes well.
+    latest_root_metadata_file = self._get_file(
```

Comment:
By moving from `_get_metadata_file` to `_get_file` you are "bypassing" the following checks on the downloaded root metadata file:
-  is [`spec_version` field](https://github.com/theupdateframework/tuf/blob/c75a2b7fe1eaabc88f82a5f0bdfc8bd25cab7ab1/tuf/client/updater.py#L1519-L1520) properly formatted,
- is [major `spec_version` supported](https://github.com/theupdateframework/tuf/blob/c75a2b7fe1eaabc88f82a5f0bdfc8bd25cab7ab1/tuf/client/updater.py#L1513-L1516),
- if there is an existing root file, is [current metadata version lower](https://github.com/theupdateframework/tuf/blob/c75a2b7fe1eaabc88f82a5f0bdfc8bd25cab7ab1/tuf/client/updater.py#L1548-L1550) than of downloaded root file, and
- via [`_verify_uncrompressed_metadata_file`](https://github.com/theupdateframework/tuf/blob/c75a2b7fe1eaabc88f82a5f0bdfc8bd25cab7ab1/tuf/client/updater.py#L1555):
  - is [metadata gernally properly formatted](https://github.com/theupdateframework/tuf/blob/c75a2b7fe1eaabc88f82a5f0bdfc8bd25cab7ab1/tuf/client/updater.py#L1428),
  - is  [metadata  not expired](https://github.com/theupdateframework/tuf/blob/c75a2b7fe1eaabc88f82a5f0bdfc8bd25cab7ab1/tuf/client/updater.py#L1431), and
  - does [metadata have good signatures](https://github.com/theupdateframework/tuf/blob/c75a2b7fe1eaabc88f82a5f0bdfc8bd25cab7ab1/tuf/client/updater.py#L1438-L1442).

Is this intended? I don't see where these checks are made up for. Is the chain of trusted root files, as described in the docstring of `_update_root_metadata`, established somewhere else?

Reply:
Certainly we don't want to check for expiry and good signatures right now, because let's say you're at v2, you don't have v3 yet, and the latest is v4. You just want to find out what the latest version is (v4), so that you rotate from v2->v3, then v3->v4, performing the metadata expiry and signatures on each rotation.

There might be some other useful checks we can do, such as checking `spec_version`, but the thing we want to _first_ is to get the _latest_ root _version number_. Does this make sense?